### ISSUE TITLE
[New Rule] The use of literal logical operators MUST be avoided

### ISSUE THREAD

**Issue by lenaorobei** [magento/magento-coding-standard#23]
### Rule
The use of literal logical operators (`and`, `or`) is discouraged. `&&` and `||` should be used instead.

### Reason
[Operator precedence](http://php.net/manual/en/language.operators.precedence.php) is different for `and` vs. `&&` and `or` vs. `||`. 
It may cause unexpected behaviour:
```php
$x = false || true;  // is ($x = (false || true)) => true
$x = false or true;  // is (($x = false) or true) => false
    
$x = true && false;  // is ($x = (true && false)) => false
$x = true and false; // is (($x = true) and false) => true
```
Please use the [official documentation](http://php.net/manual/en/language.operators.logical.php) for more details.

### Implementation
Add `Squiz.Operators.ValidLogicalOperators` to the `ruleset.xml`.

**Comment by lenaorobei** [comment]
Fixed #40
